Jenkinsfile

```diff
@@ -1,91 +1,117 @@
+          name: 'TEST_REPO'
+        ),
+        string(
+          defaultValue: 'tests',
+          description: 'Directory containing tests to run. Should at least one of the follow: an ansible-playbooks directory containing one or more test directories each of which having a playbook.yml, a scripts directory containing one or more test directories each of which having a run-test.sh',
+          name: 'TEST_DIR'
+        )
+      ]
+    )
+  ]
 )
 
 List arches = params.ARCHES.tokenize(',')
 
 library changelog: false,
-        identifier: "multiarch-ci-libraries@${params.LIBRARIES_BRANCH}",
-        retriever: legacySCM([$class: 'GitSCM', branches: [[name: "*/${params.LIBRARIES_BRANCH}"]],
-                      doGenerateSubmoduleConfigurations: false,
-                      extensions: [], submoduleCfg: [],
-                      userRemoteConfigs:
-                              [[url: "${params.LIBRARIES_REPO}"]]])
+identifier: "multiarch-ci-libraries@${params.LIBRARIES_VERSION}",
+retriever: modernSCM([$class: 'GitSCMSource',remote: "${params.LIBRARIES_REPO}"])
 
 parallelMultiArchTest(
-        arches,
-        provisioningConfig.create(params),
+  arches,
+  provisioningConfig.create(),
   { Slave slave ->
-    /*******************************************************************/
-    /* TEST BODY                                                       */
-    /* @param slave               Name of the provisioned host.        */
-    /* @param arch                Architecture of the provisioned host */
-    /*******************************************************************/
-      dir('test') {
-          stage ('Download Test Files') {
-              checkout scm
-          }
+    /*********************************************************/
+    /* TEST BODY                                             */
+    /* @param slave               Provisioned slave details. */
+    /*********************************************************/
+    dir('test') {
+      stage ('Download Test Files') {
+        if (params.TEST_REPO) {
+          git url: params.TEST_REPO, branch: params.TEST_VERSION, changelog: false
+        }
+        else {
+          checkout scm
+        }
+      }
 
-          // TODO insert test body here
-          stage ('Run Test') {
-              if (config.runOnSlave) {
-                  sh 'ansible-playbook tests/ansible-playbooks/*/playbook.yml'
-              } else {
-                  sh "ansible-playbook -i \'${slave.inventory}\' tests/ansible-playbooks/*/playbook.yml"
-              }
-          }
+      // TODO insert test body here
+      stage ('Run Test') {
+        if (config.runOnSlave) {
+          sh "ansible-playbook ${TEST_DIR}/ansible-playbooks/*/playbook.yml"
+          // TODO insert logic for calling script(s) here
+        }
+        else {
+          sh "ansible-playbook -i \'../${slave.inventory}\' ${TEST_DIR}/ansible-playbooks/*/playbook.yml"
```

Comment:
should be \'/${slave.inventory}\'
that path is absolute.

Reply:
does it need the leading slash if we're expected an absolute path slave inventory?
Shouldn't it be just be `'${slave.inventory}' ` ?

Reply:
yes. sorry typo :)